docs/android_sqlite/android_paging.md

```diff
@@ -24,6 +26,100 @@ LIMIT :limit OFFSET :offset;
+
+The `queryProvider` callback that this paging source accepts has two parameters -- a `beginInclusive` non-null unique `Key` as well as an `endExclusive` nullable unique `Key?`. An example of the core paging query is shown below. 
+
+```sql
+keyedQuery:
+SELECT * FROM hockeyPlayer
+WHERE id >= :beginInclusive AND (value < :endExclusive OR :endExclusive IS NULL)
+ORDER BY id ASC;
+```
+
+Queries used in keyset paging must have a unique ordering like shown above. 
+
+Both `beginInclusive` and `endExclusive` are _pre-calculated_ keys that act as page boundaries. Calculating the page boundaries is where page sizes are established. The `pageBoundariesProvider` callback takes an `anchor: Key?` parameter as well as a `limit: Long?` parameter. An example query that pre-calculates page boundaries is shown below. 
+
+```sql
+pageBoundaries:
+SELECT id 
+FROM (
+  SELECT
+    id,
+    CASE
+      WHEN ((row_number() OVER(ORDER BY id ASC) - 0) % :limit) = 0 THEN 1
+      WHEN id = :anchor THEN 1
+      ELSE 0
+    END page_boundary;
+  FROM hockeyPlayer
+  ORDER BY id ASC
+)
+WHERE page_boundary = 1;
+```
+
+Pre-calculating page boundaries of a SQL query will likely require [SQLite Window Functions](https://www.sqlite.org/windowfunctions.html). Window functions were introduced in SQLite version 3.25.0, and therefore are not available by default until Android API 30. To use keyset paging SQLDelight recommends either setting `minApi 30` _or_ bundling your own SQLite version. The Requery organization [offers an up-to-date distribution](https://github.com/requery/sqlite-android) of SQLite as a standalone library. 
+
+The AndroidX paging library allows for the first page fetch to differ in size from the subsequent page fetches with `PagingConfig.initialLoadSize`. This functionality should be avoided, as the `pageBoundariesProvider` callback is invoked a single time on the first page fetch. Failing to have matching `PagingConifg.initialLoadSize` and `PagingConfig.pageSize` will result in unexpected page boundary generation. 
+
+This paging source _does not_ support jumping. 
+
+To create this paging source, use the `QueryPagingSource` factory function. 
+
+```kotlin
+import com.squareup.sqldelight.android.paging3.QueryPagingSource
+
+val keyedSource = QueryPagingSource(
+  transacter = playerQueries,
+  dispatcher = Dispatchers.IO,
+  pageBoundariesProvider = playerQueries::pageBoundaries,
+  queryProvider = playerQueries::keyedQuery,
+)
+```
+
+By default, queries are performed on `Dispatchers.IO` if no dispatcher is specified. Consumers expecting to use RxJava's `Scheduler` to perform queries on should use the [`Scheduler.asCoroutineDispatcher`](https://kotlin.github.io/kotlinx.coroutines/kotlinx-coroutines-rx2/kotlinx.coroutines.rx2/io.reactivex.-scheduler/as-coroutine-dispatcher.html) extension function.
```

Comment:
Same typo/awkwardness with "queries on should" as above.

Reply:
resolved!